baus/models.py

```diff
@@ -296,6 +302,43 @@ def jobs_relocation(jobs, employment_relocation_rates, years_per_iter,
                                 pd.Series(-1, index=index))
 
 
+@orca.step()
+def household_relocation(households, household_relocation_rates,
+                         settings, static_parcels, buildings):
+
+    # get buildings that are on those parcels
+    static_buildings = buildings.index[
+        buildings.parcel_id.isin(static_parcels)]
+
+    hh_rr = household_relocation_rates.local.copy()
+    hh_rr.loc[hh_rr.tenure == 1, 'tenure'] = 'own'
+    hh_rr.loc[hh_rr.tenure == 2, 'tenure'] = 'rent'
+
+    df = pd.merge(households.to_frame(["zone_id", "base_income_quartile",
+                                       "tenure"]), hh_rr,
+                  on=["zone_id", "base_income_quartile", "tenure"],
+                  how="left")
+
+    df.index = households.index
+
+    # get the move rate for each job
+    rate = df.rate
+    # get random floats and move households if they're less than the rate
+    move = np.random.random(len(rate)) < rate
+
+    # also don't move households that are on static parcels
+    move &= ~households.building_id.isin(static_buildings)
+
+    # get the index of the moving jobs
+    index = households.index[move]
+    print("{} households are relocating".format(len(index)))
```

Comment:
@mkreilly Are we using python3 yet?  The one feature that has blown me away is [f strings](https://realpython.com/python-f-strings/) so if we're on python3 I'd recommend using that instead of `.format`

Reply:
Not using python3 yet but had better get there soon.